Python/peephole.c

```diff
@@ -475,6 +475,14 @@ PyCode_Optimize(PyObject *code, PyObject* consts, PyObject *names,
     CONST_STACK_CREATE();
 
     for (i=find_op(codestr, 0) ; i<codelen ; i=nexti) {
+        /* Bail out if an exception is set
+         * bpo-25828: fold_binops_on_constants may stuck when there is a big
+         * nubmer to calculate, e.g. 2 ** (2 ** 100). It may need to handle
+         * KeyboardInterrupt or MemoryError
+         */
+        if (PyErr_Occurred())
```

Comment:
Calling `PyErr_Occurred()` on every instruction can significantly slow down compilation. It should be called only after executing operations that can raise an exception.